crates/weaver_semconv/src/attribute.rs

```diff
@@ -429,6 +430,59 @@ pub enum Examples {
     ListOfStrings(Vec<Vec<String>>),
 }
 
+impl Examples {
+    /// Validation logic for the group.
+    pub(crate) fn validate(
+        &self,
+        future_mode: bool,
+        attr_type: &AttributeType,
+        group_id: &str,
+        attr_id: &str,
+        path_or_url: &str,
+    ) -> Result<(), Error> {
+        match (self, attr_type) {
+            (Examples::Bool(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Boolean))
+            | (Examples::Int(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Int))
+            | (Examples::Double(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Double))
+            | (Examples::String(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::String))
+            | (Examples::Ints(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Int))
+            | (Examples::Doubles(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Double))
+            | (Examples::Bools(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Boolean))
+            | (Examples::Strings(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::String))
+            | (Examples::ListOfInts(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Ints))
+            | (Examples::ListOfDoubles(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Doubles))
+            | (Examples::ListOfBools(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Booleans))
+            | (Examples::ListOfStrings(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Strings)) => {
+                Ok(())
+            }
+            (_, Enum { .. }) => {
+                // enum types are open so it's not possible to validate the examples
+                Ok(())
+            }
+            // Only if future mode is disabled, we allow to have examples following
+            // the conventions used in semconv 1.27.0 and earlier.
+            (Examples::Ints(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Ints))
+            | (Examples::Doubles(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Doubles))
+            | (Examples::Bools(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Booleans))
+            | (Examples::Strings(_), PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Strings))
+                if !future_mode =>
```

Comment:
Another alternative, instead of wiring future mode down into every single module, is to find a way to put it on diagnostics and elevate warnings into errors.

Sadly - We don't really have a result type that allows this today, and we'd need to spend more time on the ole diagnostic framework setup.

For now this is fine, we can expand our capabilities if `--future` winds up needing more central features and controls.

Reply:
I believe it’s actually possible with the existing diagnostic framework. I’ll try to implement it before the merge.

Reply:
I have completely restructured the implementation of this flag. Now, the code generates Rust warnings for items that were acceptable in the previous version of the semantic conventions but are considered errors when the `--future` flag is enabled. This change globally converts warnings into errors. The CLI’s error code has also been updated to reflect these changes.
Please review the modifications.